tests/unit/test_fw_utils.py

```diff
@@ -93,6 +94,40 @@ def test_tar_and_upload_dir_s3_with_kms(utils, sagemaker_session):
     obj.upload_file.assert_called_with(utils.create_tar_file(), ExtraArgs=extra_args)
 
 
+@patch("sagemaker.utils")
+def test_tar_and_upload_dir_s3_without_kms_no_settings(utils, sagemaker_session):
```

Comment:
nit: should this be `test_tar_and_upload_dir_s3_with_kms_no_settings` ?

Reply:
here I'm testing that even when we don't pass a `kms_key` argument the artifact uploaded to `S3` is encrypted at rest with the default AWS managed KMS key for `S3`.

How about renaming `test_tar_and_upload_dir_s3_no_custom_kms_no_settings`?